support/ccm.rb

```diff
@@ -854,13 +854,23 @@ def cluster_name
   def cassandra_version
     parse_version
     @cassandra_version ||= begin
-      version = @raw_version
-      if (@raw_version.start_with?('4.0') || @raw_version.start_with?('4.5') || @raw_version.start_with?('4.6')) and @dse
-        version = '2.0.17'
-      elsif (@raw_version.start_with?('4.7') || @raw_version.start_with?('4.8')) and @dse
-        version = '2.1.12'
+      if @dse
+        return case
+               when @raw_version =~ /4.[056].*/
+                 '2.0'
+               when @raw_version =~ /4.[78].*/
+                 '2.1'
```

Comment:
There is actually quite a bit of variation in the C* patch version within a given major/minor line of DSE.  The C* major/minor stays fairly constant, however, so to avoid the arbitrary selection of some patch version that was happening in the old code it seemed more correct to just use major/minor here.